tests/math/test_is_abstract.py

```diff
@@ -150,9 +150,13 @@ def cost(x, w, w_is_abstract=True):
         res = cost(x, w)
         assert res == 0.26
 
-        # since we only specify argnums=0, w will not be abstract
-        grad = jax.grad(cost, argnums=0)(x, w, w_is_abstract=False)
-        assert np.allclose(grad, 2 * x)
+        # NOTE: As of JAX 0.3.1, JAX will trace non-differentiated arguments.
+        # As a result, it is no longer possible to assume that non-differentiated
+        # arguments will not be abstract.
+
+        # # since we only specify argnums=0, w will not be abstract
+        # grad = jax.grad(cost, argnums=0)(x, w, w_is_abstract=False)
+        # assert np.allclose(grad, 2 * x)

```

Comment:
Just a minor thought: instead of leaving uncommented code in the test, isn't it better to just delete this part?

Reply:
Yeah I tossed it around in my head, but, I thought this might be good 'institutional knowledge' to keep for core developers, and I think this might be the best case for it 🤔 At least, looking at the source code, it doesn't really fit anywhere there --- this behaviour is best shown via example